superset-frontend/src/explore/components/controls/SelectAsyncControl/index.tsx

```diff
@@ -97,10 +100,12 @@ const SelectAsyncControl = ({
         endpoint: dataEndpoint,
       })
         .then(response => {
-          const data = mutator
-            ? mutator(response.json, value)
-            : response.json.result;
-          setOptions(data);
+          if (value) {
```

Comment:
@msyavuz not sure if this got missed but it seems we might still ignoring `null` values with this check which should be valid.